tests/ut/test_base.py

```diff
@@ -231,6 +231,93 @@ def test_alt_build_key(self):
         cache = BaseCache(key_builder=lambda key, namespace: 'x')
         assert cache.build_key(pytest.KEY, 'namespace') == 'x'
 
+    @pytest.fixture
+    def set_test_ttl(self, base_cache):
+        base_cache.ttl = 10
+        yield
+        base_cache.ttl = None
+
+    @pytest.mark.asyncio
+    async def test_add_ttl_default(self, set_test_ttl, base_cache):
+        base_cache._add = CoroutineMock()
+
+        await base_cache.add(pytest.KEY, "value")
+
+        assert base_cache._add.call_count == 1
+        assert base_cache._add.call_args[1]['ttl'] == 10
+
+    @pytest.mark.asyncio
+    async def test_add_ttl_overriden(self, set_test_ttl, base_cache):
+        base_cache._add = CoroutineMock()
+
+        await base_cache.add(pytest.KEY, "value", ttl=20)
+
+        assert base_cache._add.call_count == 1
+        assert base_cache._add.call_args[1]['ttl'] == 20
+
+    @pytest.mark.asyncio
+    async def test_add_ttl_none(self, set_test_ttl, base_cache):
+        base_cache._add = CoroutineMock()
+
+        await base_cache.add(pytest.KEY, "value", ttl=None)
+
+        assert base_cache._add.call_count == 1
+        assert base_cache._add.call_args[1]['ttl'] is None
```

Comment:
I think we are missing the case where we don't pass `ttl` explicitly and its not set either at he class too

Reply:
Added